src/python/pants/build_graph/target_filter_subsystem.py

```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from pants.subsystem.subsystem import Subsystem
+
+
+logger = logging.getLogger(__name__)
+
+
+class TargetFilter(Subsystem):
+  """Filter out targets matching given options.
+
+  :API: public
+  """
+
+  options_scope = 'target-filter'
+
+  @classmethod
+  def register_options(cls, register):
+    super(TargetFilter, cls).register_options(register)
+
+    register('--exclude-tags', type=list,
+             default=[],
+             help='Skip targets with given tag(s).')
+
+  def apply(self, targets):
+    exclude_tags = set(self.get_options().exclude_tags)
+    return TargetFiltering.apply_tag_blacklist(exclude_tags, targets)
+
+
+class TargetFiltering(object):
```

Comment:
It appears this was broken out primarily to make tests lighter weight. It's a bit odd that the thing broken out was fapp a function but it's packaged up as a class. What do you think about either:
1. Sticking with class as the vehicle, but making it pull it's weight with state - taking exclude_tags in its constructor.
2. Just inline the logic into TargetFilter and have the tests be slightly heavier wieght, with amelioration coming through a tiny bit of test infra so that the existing test methods read about as cleanly as they do now.

Reply:
👍Sounds good, I'm leaning towards #1 here to help keep things straightforward for tests and open for additional options down the line should any be added.